marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/mapping-functions/core.sjs

```diff
@@ -91,8 +91,20 @@ function parseDate(value, pattern) {
   if (pattern != null) {
     pattern = pattern.toString().replace(", ", ",");
   }
-  if (value != null) {
+  //return in case 'value' is null or undefined
```

Comment:
I recommend removing these comments. They don't add anything - the code is clear in that if the value is null or whitespace, then null will be returned.